File: assume/app/src/androidTest/java/com/sqisland/espresso/assume/MainActivityTest.java
package com.sqisland.espresso.assume;

import android.support.test.rule.ActivityTestRule;
import android.support.test.runner.AndroidJUnit4;

import org.junit.Assume;
import org.junit.Rule;
import org.junit.Test;
import org.junit.runner.RunWith;

import java.util.Locale;

import static android.support.test.espresso.Espresso.onView;
import static android.support.test.espresso.assertion.ViewAssertions.matches;
import static android.support.test.espresso.matcher.ViewMatchers.isEnabled;
import static android.support.test.espresso.matcher.ViewMatchers.withId;

@RunWith(AndroidJUnit4.class)
public class MainActivityTest {
  @Rule
  public ActivityTestRule<MainActivity> activityRule = new ActivityTestRule<>(MainActivity.class);

  @Test
  public void buttonShouldBeEnabledInUK() {
    Assume.assumeTrue(Locale.UK.equals(Locale.getDefault()));
    onView(withId(R.id.button))
        .check(matches(isEnabled()));
  }
}

File: assume/app/src/androidTest/java/com/sqisland/espresso/assume/MainActivityTest.kt
package com.sqisland.espresso.assume

import androidx.test.espresso.Espresso.onView
import androidx.test.espresso.assertion.ViewAssertions.matches
import androidx.test.espresso.matcher.ViewMatchers.isEnabled
import androidx.test.espresso.matcher.ViewMatchers.withId
import androidx.test.ext.junit.runners.AndroidJUnit4
import androidx.test.rule.ActivityTestRule
import org.junit.Assume
import org.junit.Rule
import org.junit.Test
import org.junit.runner.RunWith
import java.util.Locale

@RunWith(AndroidJUnit4::class)
class MainActivityTest {
  @get:Rule
  val activityRule = ActivityTestRule(MainActivity::class.java)

  @Test
  fun buttonShouldBeEnabledInUK() {
    Assume.assumeTrue(Locale.UK == Locale.getDefault())
    onView(withId(R.id.button))
      .check(matches(isEnabled()))
  }
}